### ISSUE TITLE
How can I use deis register from outside of a VPC?

### ISSUE THREAD

**Issue by jarcas-cdmon** [deis/deis#3781]
Hi! I've an issue when I try to execute the deis register command. I've a fresh installation in aws with the script provision-ec2-cluster.sh.
The steps I follow are:
I create a cluster of three nodes and I associate an elastic ip for each one.
I can connect to ssh to every node without problems.
I execute: 
export DEISCTL_TUNNEL= XXXXX, where XXX is an elastic IP associated to one node.
deisctl config platform set domain=XXXXX.nip.io , where XXX is the same IP than before.
deisctl config router set proxyProtocol=1
deisctl install platform
deisctl start platform
deis register http://deis.XXXXXX.nip.io

I get this error when execute deis register:

http://deis.XXXXX.nip.io does not appear to be a valid Deis controller.
Make sure that the Controller URI is correct and the server is running.
Specific Error: ('Connection aborted.', error(110, 'Connection timed out')) 

What can I do to solve this issue?

Thanks!!!

**Comment by bacongobbler** [comment]
on EC2 the only ports that are opened by the security group is port 22 for SSH/deisctl access. You;ll have to use the load balancer to connect to the nodes.

**Comment by jarcas-cdmon** [comment]
When I make a curl to the ELB from the client I get a timeout, but if I make the curl from of the nodes it has response.
It's a normal behaviour?

**Comment by bacongobbler** [comment]
That depends on how you're curl'ing the nodes and the ELB, along with the responses you're seeing. Would you please provide a gist so we can better understand your issue? I'm mostly interested in the responses so I can tell what you're seeing on your end. :)

**Comment by jarcas-cdmon** [comment]
Hi! I've changed some parameters of my configuration because I saw in the documentation:
"Note that xip does not seem to work for EC2 ELBs - you will have to use an actual DNS record."
So I changed the domain to a hosted name created with route 53 that has a CNAME of the ELB address.
However, after apply these changes and restart the platform I have the same issue, can't use the "deis register" command. I get this error:

 [root@ip-172-31-20-167 deis]# deis register omykuji.com

http://omykuji.com does not appear to be a valid Deis controller.
Make sure that the Controller URI is correct and the server is running.

Specific Error: ('Connection aborted.', gaierror(-2, 'Name or service not known'))


As I said before, I can't make a curl from the client to the ELB, it gives a timeout. The gateway and subnets seems to be ok.

bacongobbler, you ask me for a gist, but I don't see relevant information to show you. With a deisctl list I see all services up and I can connect via ssh to the nodes. Maybe you know some command to debug Deis? 


Thanksss

**Comment by bacongobbler** [comment]
> deis register omykuji.com

You need to use `deis.omykuji.com` to connect to the controller. The router forwards requests from the `deis` subdomain to the controller, but root TLD requests will just hit the default page.

> bacongobbler, you ask me for a gist, but I don't see relevant information to show you. Maybe you know some command to debug Deis?

All I needed was the input/output of `deis register`, which you just showed me. :)

**Comment by jarcas-cdmon** [comment]
I'm sorry, I writed wrong the command I use, it is:
 deis register http://deis.omykuji.com
But it also give an error
: (

 deis register http://deis.omykuji.com

http://deis.omykuji.com does not appear to be a valid Deis controller.
Make sure that the Controller URI is correct and the server is running.

Specific Error: ('Connection aborted.', gaierror(-2, 'Name or service not known'))

**Comment by bacongobbler** [comment]
Okay. `name or service not known` means that it's a DNS issue. From my own laptop, it looks like the routes have not been applied yet.

```
$ dig deis.omykuji.com

; <<>> DiG 9.8.3-P1 <<>> deis.omykuji.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 63486
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;deis.omykuji.com.		IN	A

;; Query time: 39 msec
;; SERVER: 192.168.0.1#53(192.168.0.1)
;; WHEN: Tue Jun  2 14:18:03 2015
;; MSG SIZE  rcvd: 34
```

**Comment by carmstrong** [comment]
@jarcas-cdmon Your DNS records are not properly configured:
```console
$ dig +short deis.omykuji.com
```
Returns no records. See http://docs.deis.io/en/latest/managing_deis/configure-dns/#necessary-dns-records

**Comment by mboersma** [comment]
@jarcas-cdmon either the DNS isn't set up correctly, or the Deis components aren't actually running. Did you complete `deisctl install platform && deisctl start platform`? And here's a screenshot of a working route53 config just for reference--this works for `deis register http://deis.test170.deis.works/`.

![screen shot 2015-06-02 at 3 19 15 pm](https://cloud.githubusercontent.com/assets/73019/7947565/20a1f7e8-093b-11e5-8ce4-25c0246e4e34.png)

**Comment by jarcas-cdmon** [comment]
Hey! I have configured a private hosted zone in Route 53 to route the traffic of xxxx.omykuji.com to the ELB, for this reason you get no records in this domain, this configuration is only available for the servers I have in my zone. Maybe this configuration can be a problem for Deis, so I will try to change it.

@mboersma I complete the deisctl install platform & deisctl start platform without problems, but I remember when I executed the script provision-ec2 a warning saying that I'd to up the proxy service manually, I executed this command:
deisctl config router set proxyProtocol=1
It's all correct?

**Comment by carmstrong** [comment]
> Maybe this configuration can be a problem for Deis, so I will try to change it.

That should be fine, as long as the host the `deis` client is on can also resolve those records. From your error message, it appears that it cannot.

> It's all correct?

Yes, that sounds correct. The error coming from your workstation is definitely a DNS issue, so try `dig deis.omykuji.com` and I'm betting you don't see the records.

**Comment by jarcas-cdmon** [comment]
Hey! I've created a new domain that has a CNAME of my ELB. It is: wearjarc.cat
I've make a new installation of Deis and set:
deisctl config platform set domain=wearjarc.cat

But when I run the register command it gives a timeout as before...
deis register http://deis.wearjarc.cat
With dig +short deis.wearjarc.cat I can see how it redirects correctly.

After this, I created a new instance in the same VPC where the nodes are located to run Deis register. And ... It works!!
However, I want to use Deis client with an instance out of this VPC....

In the documentation I found this parameter:
export http_proxy="http://proxyip:port"
I'm not sure if I have to set a value for this parameter?

Thanks!!

**Comment by bacongobbler** [comment]
The only information I could find on connecting to a VPC was [this](http://docs.aws.amazon.com/AmazonElastiCache/latest/UserGuide/ManagingVPC.Connecting.html), which is basically what you just did. I'm not sure if there's another way, but maybe others who are more familiar with VPC could shed more light.

**Comment by jarcas-cdmon** [comment]
Hi! 
In  the cloudformation file I've found this param:
"ParameterKey":     "ELBScheme",
    "ParameterValue":   "internal"

For this reason the ELB only worked with the private subnet.
I copied this parameter from the documentation, as an example of the cloudformation: http://docs.deis.io/en/latest/installing_deis/aws/#customize-cloudformation-json 
Maybe it's a bit confusing if it's the first time that someone install Deis.
Thanks for all!
